### PR TITLE
Update version of windows_exporter

### PR DESCRIPTION
**Pull request by MarkDordoy**
#### PR Description
This PR updates the windows_exporter dependency to master (commit hash f8c298e038f8e0fcaf7f77fd591a5e1dfeb7f27b). Full change log [here](https://github.com/prometheus-community/windows_exporter/compare/79781c6d75fc..f8c298e038f8)

#### Notes to the Reviewer

 - `windows_ad_ldap_client_sessions` added to ad collector. (@MarkDordoy)
  - `windows_hyperv_host_cpu_wait_time_per_dispatch_total` and `windows_hyperv_vm_cpu_wait_time_per_dispatch_total` added to hyperv collector. (@yuriyostapenko)
  - nps collector created. (@rebortg)
  - adfs collector metric docs updated to correctly reflect metric names. (@breed808)

#### PR Checklist

- [x] CHANGELOG.md updated

### CODE REVIEW

**Comment by MarkDordoy** [comment]
I think this is ready for review. My first time contributing so please let me know if anything is missing/required and I'll get this updated.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ Main (unreleased)
 ### Breaking changes
 
 - Set `retry_on_http_429` to `true` by default in the `queue_config` block in static mode's `remote_write`. (@wildum)
+- windows_exporter adfs collector update.
```
Comment (tpaschalis) [inline CHANGELOG.md:16]: Is this a breaking change really? 

If so, once we're done upgrading our dependency properly, we should add some details around _why_ is this a breaking change. Any new or removed config fields should be accounted for in documentation and should also be reflected in the two `release-notes.md` documents

Comment (MarkDordoy) [reply CHANGELOG.md]: apologies, i put that in originally thinking it was, but it turns out it was just bringing the doc's inline with the underlying code. https://github.com/prometheus-community/windows_exporter/pull/1272

I've updated the CHANGELOG file

File: go.mod [hunk 2]
```diff
@@ -134,7 +134,7 @@ require (
 	github.com/prometheus-community/elasticsearch_exporter v1.5.0
 	github.com/prometheus-community/postgres_exporter v0.11.1
 	github.com/prometheus-community/stackdriver_exporter v0.13.0
-	github.com/prometheus-community/windows_exporter v0.0.0-20230507104622-79781c6d75fc
+	github.com/prometheus-community/windows_exporter v0.23.2-0.20230904201428-f8c298e038f8
```
[the remark this answers is not in the document]
Comment (mattdurham) [reply go.mod:137]: Its on my todo, I have a PR to but it got stale. https://github.com/prometheus-community/windows_exporter/pull/1227

will see if i can find some time to resolve

[the remark this answers is not in the document]
Comment (MarkDordoy) [reply go.mod:137]: for my future reference @mattdurham , if I want to make changes/extend the windows_exporter, should I update the prometheus or grafana repo?

[the remark this answers is not in the document]
Comment (mattdurham) [reply go.mod:137]: Grafana though at some point we will get the changes merged upstream.

[the remark this answers is not in the document]
Comment (MarkDordoy) [reply go.mod:137]: @mattdurham ok cool, for now I've added the update into the grafana fork, PR here: https://github.com/grafana/windows_exporter/pull/21 

if/when you merge that, I'll create a new PR to update the commit in the agent. 

Thanks

**Comment by mattdurham** [comment]
closing this since we have updated windows exporter and no longer need the grafana fork

**Pull request closed without merging**